src/Polly.Core/Retry/RetryHelper.cs

```diff
@@ -0,0 +1,26 @@
+using System;
+
+namespace Polly.Retry;
+
+internal static class RetryHelper
+{
+    private const double ExponentialFactor = 2.0;
+
+    public static bool IsValidDelay(TimeSpan delay) => delay >= TimeSpan.Zero;
+
+    public static TimeSpan GetRetryDelay(RetryBackoffType type, int attempt, TimeSpan baseDelay)
+    {
+        if (baseDelay == TimeSpan.Zero)
+        {
+            return baseDelay;
+        }
+
+        return type switch
+        {
+            RetryBackoffType.Constant => baseDelay,
+            RetryBackoffType.Linear => TimeSpan.FromMilliseconds((attempt + 1) * baseDelay.TotalMilliseconds),
+            RetryBackoffType.Exponential => TimeSpan.FromMilliseconds(Math.Pow(ExponentialFactor, attempt) * baseDelay.TotalMilliseconds),
```

Comment:
Does `TimeSpan` not have the built-in operators to do them like this?

```suggestion
            RetryBackoffType.Linear => (attempt + 1) * baseDelay,
            RetryBackoffType.Exponential => Math.Pow(ExponentialFactor, attempt) * baseDelay,
```

Reply:
Sadly no at least on .NET Framework. I can put this behind ifdefs.

Reply:
Ah OK - let's leave it as-is then.